app_server/test/services/UserService.js

```diff
@@ -85,6 +85,24 @@ lab.experiment('UserService tests', function () {
     });
   });
 
+  lab.test('getUserById valid arguments', function(done) {
+    UserService.createNewUser(bob).then(function (result) {
+      return UserService.getUserById(result.userId);
+    }).then(function(user) {
+      Code.expect(user.username).to.equal(bob.username);
+      Code.expect(user.password).to.equal(bob.password);
+      done();
+    });
+  });
+
+  lab.test('getUserById invalid arguments', function(done) {
```

Comment:
Ideally, methods in `Service`should be tested using verification (verify that it calls `UserService` once and receive corresponding result) so that it doesn't need to run through the whole flow but still can confirm the link between `Service` and `UserService` . But that's quite messy to implement so I think duplicated test in `UserService` can be removed for now as they are supposed to be internal subcomponents of `Service`.